airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/write/Batch.kt

```diff
@@ -0,0 +1,14 @@
+package io.airbyte.cdk.write
+
+/**
+ * Represents an accumulated batch of records in some stage of processing.
+ *
+ * Emitted by the record accumulator per batch accumulated. Handled by
+ * the batch processor, which may advanced the state and yield a new batch.
+ *
+ * If persisted is true, the framework will close state for the accumulated messages.
+ */
+open class Batch(
+    val name: String,
```

Comment:
I see the names as id's, and you can either use them to store file/table names, or you can extend batch to add more useful purpose-specific metadata

Reply:
got it, that makes sense. Do you have a strong preference for inheritance vs just doing `Batch<T>(..., val data: T)`? I mildly prefer it, if only to avoid having very boilerplatey classes like this (i.e. where the only interesting thing is `val file: File`)
```kotlin
class LocalFileBatch(name, persisted, val file: File): Batch(name, persisted)
```

but haven't thought through this much so maybe there are weird consequences to this

Reply:
No preferences, beyond just avoiding propagating generics everywhere.